rcsbapi/search/search_query.py

```diff
@@ -0,0 +1,1852 @@
+        start = 0
+        req_count = 0
+        response = self._single_query(start=start)
+        if response is None:
+            return  # be explicit for mypy
+        if "result_set" in response:
+            result_set = response["result_set"]
+        elif "group_set" in response:
+            result_set = response["group_set"]
+        else:
+            result_set = []
+        start += self.rows
+        logging.debug("Got %s ids", len(result_set))
+
+        if len(result_set) == 0:
+            return
+        yield from result_set
+
+        total = response["total_count"]
+
+        while start < total:
+            # If no grouping is applied, check that result_set = rows
+            # If grouping is applied, result set could be lower than rows
+            if not self._group_by:
+                assert len(result_set) == self.rows
+            req_count += 1
+            if req_count == Const.REQUESTS_PER_SECOND.value:
+                time.sleep(1.2)  # This prevents the user from bottlenecking the server with requests.
+                req_count = 0
+            response = self._single_query(start=start)
+            assert isinstance(response, dict)
+            if "result_set" in response:
+                result_set = response["result_set"]
+            elif "group_set" in response:
+                result_set = response["group_set"]
+            else:
+                result_set = []
+            logging.debug("Got %s ids", len(result_set))
+            start += self.rows
+            yield from result_set
+
+    def to_dict(self) -> Dict:
+        """return full json response"""
+        response = self._single_query()
+        if not isinstance(response, Dict):
+            return {}
+        return response
+
+    def iquery(self, limit: Optional[int] = None) -> List[str]:
+        """Evaluate the query and display an interactive progress bar.
+
+        Requires tqdm.
+        """
+        from tqdm import trange  # type: ignore
+
+        response = self._single_query(start=0)
+        if response is None:
+            return []
+        total = response["total_count"]
+        result_set = response["result_set"] if response else []
+        if limit is not None and len(result_set) >= limit:
+            return result_set[:limit]
+
+        pages = math.ceil((total if limit is None else min(total, limit)) / self.rows)
+
+        for page in trange(1, pages, initial=1, total=pages):
+            response = self._single_query(page * self.rows)
+            next_results = response["result_set"] if response else []
+            result_set.extend(next_results)
+
+        return result_set[:limit]
+
+    def rcsb_query_editor_url(self) -> str:
+        """URL to edit this query in the RCSB PDB query editor"""
+        data = json.dumps(self._make_params(), separators=(",", ":"))
+        return f"https://search.rcsb.org/query-editor.html?json={urllib.parse.quote(data)}"
+
+    def rcsb_query_builder_url(self) -> str:
```

Comment:
This method could be left alone or renamed to be more consistent like `get_query_builder_link`